cql3/statements/drop_type_statement.cc

```diff
@@ -122,21 +136,20 @@ const sstring& drop_type_statement::keyspace() const
 
 future<std::tuple<::shared_ptr<cql_transport::event::schema_change>, std::vector<mutation>, cql3::cql_warnings_vec>>
 drop_type_statement::prepare_schema_mutations(query_processor& qp, api::timestamp_type ts) const {
-    validate_while_executing(qp);
+    ::shared_ptr<cql_transport::event::schema_change> ret;
+    std::vector<mutation> m;
 
-    data_dictionary::database db = qp.db();
+    if (!no_type_when_if_exists(qp)) {
```

Comment:
Style note:
It was super hard for me to understand this double-negative, plus the combination of the logic with if_exists. I'm not even sure I followed it completely. I think everything would have been clearer without the function and just inline the code as it was. Moreover, with this extra function, you now have the all_types.find() stuff done twice which is just unnecessary duplication (of code and runtime).

Moreover, I think the logic would be clearer if this function simply returned early if there is nothing to do (e.g., when it notices that _if_exists is on but the keyspace is missing), instead of having a single ret at the end.

Reply:
@nyh hope new solution is better. Please review it.